tool/ood-gen/lib/event.ml

```diff
@@ -141,6 +141,36 @@ let all () =
          in
          String.compare t2 t1)
 
+  let create_entry (log : t) =
+    let authors = (Syndic.Atom.author "Ocaml.org", []) in
+    let event_type = EventType.show log.event_type in
+    let textual_location = log.textual_location in
+    let start_date =
+      Syndic.Date.of_rfc3339
+        (log.starts.yyyy_mm_dd ^ "T"
+        ^ Option.value ~default:"00:00" log.starts.utc_hh_mm
+        ^ ":00Z")
+    in
+    let start_date_str = Syndic.Date.to_rfc3339 start_date in
+    let id = Uri.of_string (log.slug ^ " " ^ start_date_str) in
+    let location_summary =
+      match log.location with
+      | Some { lat; long } ->
+          Printf.sprintf "%s (lat: %f, long: %f)" textual_location lat long
+      | None -> textual_location
+    in
+    Syndic.Atom.entry ~id ~authors
+      ~title:(Syndic.Atom.Text (log.title ^ " " ^ start_date_str))
+      ~updated:start_date
```

Comment:
The event start date should not be used here. The updated date is when the event announcement is created or updated.

Reply:
Oh sure, I'm in my uni rn for an exam, will check back later

Reply:
@cuihtlauac i checked the event feed markdown files, there isn't any updated date field
should i put a conditional whether when updated is present to use that or leave it as it is, thanks!

Reply:
For 1-shot events, we need an additional field in `Event.t` to hold that information. When `all` is built, using the function `decode`, we must initialize it with the file's last write time on the markdown. 

For recurring events, it's trickier. We have to check in the RSS spec if it is allowed to skip that field.

Reply:
okay

Reply:
I tried to research whether we can omit the field in recurring events or not but couldn't find anything substantial, will need some help here

Reply:
A possible solution could be to use the filesystem information for one-shot events and remove recurring events from the feed. Does it make sense to you?